plugins/handlers/gallery.py

```diff
@@ -251,27 +267,85 @@ def evaluate(self, data):
         return False
 
 
+class And(object):
+
+    def __init__(self, *tests):
+        self.tests = tests
+
+    def evaluate(self, data):
+        for test in self.tests:
+            if not test.evaluate(data):
+                return False
+        return True
+
+
 class Glob(Or):
 
     def __init__(self, glob):
         patterns = [fnmatch.translate(p) for p in braceexpand.braceexpand(glob)]
         tests = [Regex(pattern, re.IGNORECASE) for pattern in patterns]
         super().__init__(*tests)
 
+    def evaluate(self, data):
+        return super().evaluate(data["basename"])
+
+
+# TODO: Consider making this an explicit metadata key test.
+class Metadata(object):
+
+    def __init__(self, **kwargs):
+        self.kwargs = kwargs
+
+    def evaluate(self, data):
+        for key, value in self.kwargs.items():
+            metadata = data["metadata"]
+            if key not in metadata or metadata[key] != value:
+                return False
+        return True
+
 
 RESIZE_METHODS = [
     (Glob("*.gif"), gifsicle_resize),
     (Glob("*"), imagemagick_resize),
 ]
 
 
+# TODO: Remove this!!
 OUTPUT_MIME_TYPES = [
     (Glob("*.heic"), "image/jpeg"),
     (Glob("*.tiff"), "image/jpeg"),
     (Glob("*"), "*"),
 ]
 
 
+# where: glob:* and metadata:projection=equirectangular
+# then:
+# - transform: convert
+#   width: 1600
+#   format: image/jpeg
+#   tags:
+#   - image
+# - transform: convert
+#   width: 480
+#   tags:
+#   - thumbnail
+
+IMAGE_TRANSFORMS = [
+
+    (Metadata(projection="equirectangular"),
+        [Resize("large", width=10000, format="*", tags=["image"]),
+         Fisheye("preview-small", width=480, tags=["thumbnail", "previews"]),
+         Fisheye("preview-large", width=960, tags=["previews"])]),
+    (Or(Glob("*.heic"), Glob("*.tiff")),
+        [Resize("large", width=1600, format="image/jpeg", tags=["image", "previews"]),
+         Resize("small", width=480, format="image/jpeg", tags=["thumbnail", "previews"])]),
+    (Glob("*"),
+        [Resize("large", width=1600, tags=["image", "previews"]),
+         Resize("small", width=480, tags=["thumbnail", "previews"])]),
+
+]
```

Comment:
Long term, we'd want to move this out into the configuration file (see earlier comment).

Reply:
In the short term, it's probably worth reviewing the file names and tags.